autograder-api/src/main/java/de/firemage/autograder/api/AbstractLinter.java

```diff
@@ -0,0 +1,71 @@
+package de.firemage.autograder.api;
+
+import fluent.bundle.FluentBundle;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Consumer;
+
+public interface AbstractLinter {
+    List<? extends AbstractProblem> checkFile(Path file, JavaVersion version, CheckConfiguration checkConfiguration, Consumer<Translatable> statusConsumer) throws LinterException, IOException;
```

Comment:
Should we expose `JavaVersion`? Wouldn't this risk people forgetting about it and having a fixed version like `JavaVersion.JAVA_17` somewhere that would break when we update to Java 21?

And from what I can tell, one can always parse older java code with newer java versions, so it shouldn't be necessary to have this here. By the way, for the same reason, I want to remove the explicit java version set in every test as well. (Didn't do it yet, because it would touch hundreds of files)

Reply:
Not necessarily, since there may be checks (e.g. get(0) -> getFirst()) that are only applicable for specific Java versions

Reply:
The autograder is currently not capable of having checks that only work with specific java versions. It also is very unlikely that we would downgrade from like Java 21 to 17 which would necessitate this.

Who knows what we do in the future, so we could keep it. Hopefully, nobody forgets changing that when we start using newer java versions